core/app-framework/sensor/native/runtime_sensor.c

```diff
@@ -395,12 +395,12 @@ check_sensor_timers()
 
             s->last_read = now;
 
-            if (ms_to_next_check == -1 || (ms_to_next_check < s->read_interval))
+            if (s->read_interval < ms_to_next_check)
                 ms_to_next_check = s->read_interval;
         }
         else {
-            int remaining = s->read_interval - elpased_ms;
-            if (ms_to_next_check == -1 || (ms_to_next_check < remaining))
+            uint32 remaining = s->read_interval - elpased_ms;
```

Comment:
shall we be worried if `s->read_interval` is smaller than `elpased_ms` ?

Reply:
This is the else branch, s->read_interval <= elpased_ms is handled in if branch, see Line 385:
`if (elpased_ms >= s->read_interval) {`